src/App.tsx

```diff
@@ -1,61 +1,137 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+let lastId = Math.max(...todos.map(({ id }) => id));
+
+export const App: React.FC = () => {
+  const [newTodos, setNewTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUserId, setSelectedUserId] = useState('');
+  const [userHasError, setUserHasError] = useState(false);
+  const [titleHasError, setTitleHasError] = useState(false);
+
+  const getNewTodo = () => {
+    const newId = lastId + 1;
+
+    lastId += 1;
+
+    return {
+      id: newId,
+      title,
+      completed: false,
+      userId: +selectedUserId,
+      user: getUserById(+selectedUserId),
+    };
+  };
+
+  const titleInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleHasError(false);
+  };
+
+  const selectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(event.target.value);
+    setUserHasError(false);
+  };
+
+  const addNewTodo = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
The function name `addNewTodo` could be more descriptive. Consider naming it `handleAddNewTodo` to better reflect its purpose. 📝